src/files-breadcrumbs.tsx

```diff
@@ -49,19 +49,52 @@ function useHostname() {
 
 // eslint-disable-next-line max-len
 export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: string[], showHidden: boolean, setShowHidden: React.Dispatch<React.SetStateAction<boolean>>}) {
+    const [editMode, setEditMode] = React.useState(false);
+    const [newPath, setNewPath] = React.useState<string | null>(null);
     const hostname = useHostname();
 
     function navigate(n_parts: number) {
         cockpit.location.go("/", { path: encodeURIComponent(path.slice(0, n_parts).join("/")) });
     }
 
+    const handleInputKey = (event: React.KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === "Enter") {
+            changePath();
+        } else if (event.key === "Escape") {
+            cancelPathEdit();
+        }
+    };
+
+    const enableEditMode = () => {
+        setEditMode(true);
+        setNewPath(path.join("/"));
+    };
+
+    const changePath = () => {
+        setEditMode(false);
+        cockpit.assert(newPath !== null, "newPath cannot be null");
+        // HACK: strip trailing / to circumvent the path being `//` in breadcrumbs
+        cockpit.location.go("/", { path: encodeURIComponent(newPath.replace(/\/$/, '')) });
```

Comment:
same here, `setNewPath(null)` for resetting the state to non-edit mode? (Essentially the same as cancelPathEdit(), you may as well just call that one)